### ISSUE TITLE
[Bug] (crash) Using private static fields with `in` operator in itself

### ISSUE THREAD

**Issue by p51lee** [terser/terser#1300]
**Bug report**

**Version (complete output of `terser -V` or specific git commit)**
```console
$ terser -V
terser 5.15.1
```
**Complete CLI command or `minify()` options used**
I turned off the `unsafe` option.
```console
$ terser input.js -c unsafe=false
```

**`terser` input**
```js
class x { static #y = #y in x ; }
```

**`terser` output or error**
Terser throws an error:
```console
$ terser input.js -c unsafe=false
Parse error at input.js:1,22
class x { static #y = #y in x ; }
                      ^
ERROR: Unexpected token: privatename (y)
    at js_error (/opt/homebrew/lib/node_modules/terser/dist/bundle.min.js:554:11)
    at croak (/opt/homebrew/lib/node_modules/terser/dist/bundle.min.js:1278:9)
    at token_error (/opt/homebrew/lib/node_modules/terser/dist/bundle.min.js:1286:9)
    at unexpected (/opt/homebrew/lib/node_modules/terser/dist/bundle.min.js:1292:9)
    at expr_atom (/opt/homebrew/lib/node_modules/terser/dist/bundle.min.js:2531:9)
    at maybe_unary (/opt/homebrew/lib/node_modules/terser/dist/bundle.min.js:3339:19)
    at expr_ops (/opt/homebrew/lib/node_modules/terser/dist/bundle.min.js:3390:24)
    at maybe_conditional (/opt/homebrew/lib/node_modules/terser/dist/bundle.min.js:3395:20)
    at maybe_assign (/opt/homebrew/lib/node_modules/terser/dist/bundle.min.js:3472:20)
    at expression (/opt/homebrew/lib/node_modules/terser/dist/bundle.min.js:3497:24)
```

**Expected result**
The input program is expected to be terminated without error, but Terser fails to transpile it.

```console
$ node input.js # no error
```

**Comment by p51lee** [comment]
Hello, I've opened several bug reports but some of them got no response yet:

* https://github.com/terser/terser/issues/1281
* https://github.com/terser/terser/issues/1282
* https://github.com/terser/terser/issues/1283
* https://github.com/terser/terser/issues/1284
* https://github.com/terser/terser/issues/1287
* https://github.com/terser/terser/issues/1291
* https://github.com/terser/terser/issues/1292
* https://github.com/terser/terser/issues/1293
* https://github.com/terser/terser/issues/1294
* https://github.com/terser/terser/issues/1295
* https://github.com/terser/terser/issues/1296
* https://github.com/terser/terser/issues/1297
* https://github.com/terser/terser/issues/1298
* https://github.com/terser/terser/issues/1299
* https://github.com/terser/terser/issues/1300

I wonder if these issues can be considered as bugs or not. Could you kindly review these issues and give some feedback?
@fabiosantoscode @jridgewell

**Comment by fabiosantoscode** [comment]
Hey there! These reports are appreciated but this is a pretty slow project. I'm going to check them out and label appropriately.

**Comment by fabiosantoscode** [comment]
This issue is what I'm currently working on. There was a PR implementing this but I want to make sure the implementation is complete (IE works with compression, mangle, property mangle, mozilla AST)

**Comment by fabiosantoscode** [comment]
Released in 5.16.0
